package-res/resources/web/pentaho/type/model.js

```diff
@@ -0,0 +1,98 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "module",
+  "./complex",
+  "../i18n!types"
+], function(module, complexFactory, bundle) {
+
+  "use strict";
+
+  return function(context) {
+
+    var Complex = context.get(complexFactory);
+
+    /**
+     * @name pentaho.type.Model.Type
+     * @class
+     * @extends pentaho.type.Complex.Type
+     *
+     * @classDesc The base type class of model types.
+     *
+     * For more information see {@link pentaho.type.Model}.
+     */
+
+    /**
+     * @name pentaho.type.Model
+     * @class
+     * @extends pentaho.type.Complex
+     *
+     * @amd {pentaho.type.Factory<pentaho.type.Model>} pentaho/type/model
+     *
+     * @classDesc The base class of model types.
+     *
+     * @description Creates a model instance.
+     *
+     * @constructor
+     * @param {pentaho.type.spec.IModel} [spec] A model specification.
+     */
+    var Model = Complex.extend(/** @lends pentaho.type.Model# */{
+
+      // region serialization
+      toSpecInContext: function(keyArgs) {
+        if(keyArgs && keyArgs.isJson) {
+          keyArgs = keyArgs ? Object.create(keyArgs) : {};
+
+          var omitProps = keyArgs.omitProps;
+          keyArgs.omitProps = omitProps = omitProps ? Object.create(omitProps) : {};
+
+          if(omitProps.application == null) omitProps.application = true;
+        }
+
+        return this.base(keyArgs);
+      },
+      // endregion
+
+      type: /** @lends pentaho.type.Model.Type# */{
+        id: module.id,
+
+        props: [
+          /**
+           * Gets or sets the application object.
+           *
+           * The application object represents the relevant state and
+           * interface of the application in which a model is being used.
+           *
+           * This property does not serialize to JSON by default.
+           *
+           * @name application
+           * @memberOf pentaho.type.Model#
+           * @type {pentaho.type.Application}
+           */
+          {
+            name: "application",
+            type: "pentaho/type/application"
+          }
+        ]
```

Comment:
And the other fields referred in IModel.jsdoc? Or is it unrelated and I'm making confusion?